odl/tomo/geometry/conebeam.py

```diff
@@ -0,0 +1,557 @@
+
+        angle_offset = float(angle_offset)
+
+        super().__init__()
+        self._motion_params = angle_intvl
+        self._src_radius = src_rad
+        self._det_radius = det_rad
+        self._motion_grid = agrid
+        self._motion_params_offset = angle_offset
+        self._axis = axis
+
+    @property
+    def motion_params(self):
+        """Motion parameters of this geometry."""
+        return self._motion_params
+
+    @property
+    def motion_params_offset(self):
+        """Offset to motion parameters. """
+        return self._motion_params_offset
+
+    @property
+    def motion_grid(self):
+        """Sampling grid for this geometry's motion parameters."""
+        return self._motion_grid
+
+    @property
+    def angle_intvl(self):
+        """Angles (= motion parameters) of this geometry."""
+        return self._motion_params
+
+    @property
+    def angle_grid(self):
+        """Angle (= motion parameter) sampling grid of this geometry."""
+        return self._motion_grid
+
+    @property
+    def angle_offset(self):
+        """Offset to the rotation angle in the azimuthal plane. Does not
+        imply an offset in z-direction. The actual angles then reside
+        within `angle_offset` + `angle_intvl`."""
+        return self._motion_params_offset
+
+    @property
+    def src_radius(self):
+        """Source circle radius of this geometry."""
+        return self._src_radius
+
+    @property
+    def det_radius(self):
+        """Detector circle radius of this geometry."""
+        return self._det_radius
+
+    @property
+    def ndim(self):
+        """Number of dimensions of this geometry."""
+        return 3
+
+    @property
+    def axis(self):
+        """The axis of rotation.
+
+        Returns
+        -------
+        axis : `numpy.ndarray`, shape ``(3,)``
+            The rotation axis
+        """
+        axis = self._axis
+        if axis is None:
+            axis = np.array([0, 0, 1])
+        elif isinstance(axis, (int, float)):
+            ind = axis
+            axis = np.zeros(3)
+            axis[ind] = 1
+        elif len(axis) == 3:
+            axis = np.array(axis)
+        else:
+            raise ValueError('`axis` intializer {} has wrong format'.format(
+                axis))
+
+        return axis
+
+    def det_rotation(self, angle):
+        """The detector rotation function.
+
+        Returns the matrix for rotating a vector in 3d by an angle ``angle``
+        about the rotation axis given by the property `axis` according to
+        the right hand rule.
+
+        The matrix is computed according to
+        `Rodrigues' rotation formula
+        <https://en.wikipedia.org/wiki/Rodrigues'_rotation_formula>`_.
+
+        Parameters
+        ----------
+        angle : `float`
+            The motion parameter. It must be contained in this
+            geometry's `motion_params`.
+
+        Returns
+        -------
+        rot_mat : `numpy.matrix`, shape ``(3, 3)``
```

Comment:
I think we should never use `numpy.matrix` it breaks duck-typing since `*` behaves unexpectedly. Simply return an `numpy.array`.